pkg/relayer/session/claim.go

```diff
@@ -180,9 +181,26 @@ func (rs *relayerSessionsManager) newMapClaimSessionsFn(
 			return either.Success(sessionTrees), false
 		}
 
-		// Map key is the supplier operator address.
-		claimMsgs := make([]client.MsgCreateClaim, 0)
+		claimMsgs := make([]client.MsgCreateClaim, 0, len(sessionTrees))
+		// sessionTreesWithProofRequired will accumulate all the sessionTrees that
+		// will require a proof to be submitted.
+		sessionTreesWithProofRequired := make([]relayer.SessionTree, 0)
 		for _, sessionTree := range sessionTrees {
+			isProofRequired, err := rs.isProofRequired(ctx, sessionTree)
+
+			// If an error is encountered while determining if a proof is required,
```

Comment:
This is weird though, we ONLY know if a proof is required AFTER the claim was created. Otherwwise you could game the system. What am I missing?

Reply:
Grouping my reply in https://github.com/pokt-network/poktroll/pull/765#discussion_r1736436005